streams/src/test/java/org/apache/kafka/streams/integration/KTableKTableForeignKeyJoinIntegrationTest.java

```diff
@@ -0,0 +1,699 @@
+            final String two = myJoinStoreTwo.get(expected.getKey());
+            final String three = myJoinStoreThree.get(expected.getKey());
+
+            String result;
+            if (one != null)
+                result = one;
+            else if (two != null)
+                result = two;
+            else if (three != null)
+                result = three;
+            else
+                throw new RuntimeException("Cannot find key " + expected.getKey() + " in any of the state stores");
+            assertEquals(expected.getValue(), result);
+        }
+
+        //Merge all the iterators together to ensure that their sum equals the total set of expected elements.
+        final KeyValueIterator<Integer, String> allOne = myJoinStoreOne.all();
+        final KeyValueIterator<Integer, String> allTwo = myJoinStoreTwo.all();
+        final KeyValueIterator<Integer, String> allThree = myJoinStoreThree.all();
+
+        final List<KeyValue<Integer, String>> all = new LinkedList<>();
+
+        while (allOne.hasNext()) {
+            all.add(allOne.next());
+        }
+        while (allTwo.hasNext()) {
+            all.add(allTwo.next());
+        }
+        while (allThree.hasNext()) {
+            all.add(allThree.next());
+        }
+        allOne.close();
+        allTwo.close();
+        allThree.close();
+
+        for (final KeyValue<Integer, String> elem : all) {
+            assertTrue(expectedResult.contains(elem));
+        }
+    }
+
+    private KafkaStreams prepareTopology(final String queryableStoreName, final boolean leftJoin) {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final KTable<Integer, Float> left = builder.table(LEFT_TABLE, Consumed.with(Serdes.Integer(), Serdes.Float()));
+        final KTable<String, Long> right = builder.table(RIGHT_TABLE, Consumed.with(Serdes.String(), Serdes.Long()));
+
+        final Materialized<Integer, String, KeyValueStore<Bytes, byte[]>> materialized;
+        if (queryableStoreName != null) {
+            materialized = Materialized.<Integer, String, KeyValueStore<Bytes, byte[]>>as(queryableStoreName)
+                    .withKeySerde(Serdes.Integer())
+                    .withValueSerde(Serdes.String())
+                    .withCachingDisabled();
+        } else {
+            throw new RuntimeException("Current implementation of join on foreign key requires a materialized store");
+        }
+
+        if (leftJoin)
+            left.leftJoin(right, tableOneKeyExtractor, joiner, Named.as("customName"), materialized)
+                .toStream()
+                .to(OUTPUT, Produced.with(Serdes.Integer(), Serdes.String()));
+        else
+            left.join(right, tableOneKeyExtractor, joiner/*, Named.as("customName")*/, materialized)
+                .toStream()
+                .to(OUTPUT, Produced.with(Serdes.Integer(), Serdes.String()));
+
+        final Topology topology = builder.build();
```

Comment:
nit: Enabling optimizations is a two-step process.  In addition to setting the `OPTIMIZE` flag in the configs, we also need pass in the configs to the overloaded `StreamBuilder#build(Properties)` method here.

Reply:
Whoops! That's not a nit, that's just a miss on my part. Thanks!